src/Provider/Fetcher.js

```diff
@@ -104,7 +104,11 @@ export default {
             rej = reject;
         });
 
-        textureLoader.load(url, res, () => {}, rej);
+        textureLoader.load(url, res, () => {}, (event) => {
+            const error = new Error(`Failed to load texture from URL: ${url}`);
```

Comment:
Surround the url with a marker to help diagnose malformed URLs.
```suggestion
            const error = new Error(`Failed to load texture from URL: `${url}``);
```